src/rgw/rgw_admin.cc

```diff
@@ -5531,6 +5551,32 @@ int main(int argc, const char **argv)
   } /* OPT_BUCKETS_LIST */
 
   if (opt_cmd == OPT_BUCKET_STATS) {
+    if (bucket_name.empty() && !bucket_id.empty()) {
+      void *handle = NULL;
+      int ret = store->ctl()->meta.mgr->list_keys_init("bucket.instance", marker, &handle);
+      if (ret < 0) {
+        cerr << "ERROR: can't get key: " << cpp_strerror(-ret) << std::endl;
+        return -ret;
+      }
+      bool truncated = false;
+      bool bucket_from_id = false;
+      do {
+        list<string> keys;
+        ret = store->ctl()->meta.mgr->list_keys_next(handle, 1, keys, &truncated);
+        if (ret < 0) {
+          cerr << "ERROR: lists_keys_next(): " << cpp_strerror(-ret) << std::endl;
+          return -ret;
```

Comment:
in this early return, we would leak the `handle` memory allocated by `list_keys_init()` because we never call `list_keys_complete()`